src/main/scala/org/graphframes/examples/LDBCUtils.scala

```diff
@@ -0,0 +1,111 @@
+package org.graphframes.examples
+
+import org.apache.spark.sql.functions.{col, split}
+import org.apache.spark.sql.types.LongType
+import org.apache.spark.sql.{DataFrame, SparkSession}
+import org.graphframes.GraphFrame
+
+import java.net.URL
+import java.nio.file._
+import java.util.Properties
+import scala.sys.process._
+
+object LDBCUtils {
+  // TODO: This object can be actually a class, parametrized by the id of LDBC graph (kgs, etc.)
+  private val caseName = "kgs" // XS, 65.7Mb
+  private val LDBCUrl = new URL(
+    s"https://datasets.ldbcouncil.org/graphalytics/${caseName}.tar.zst")
+  private val bufferSize = 8192 // 8Kb
+  private val resourcesDir: Path = sys.env.get("LDBC_TEST_ROOT") match {
+    case Some(s) => Paths.get(s).resolve(caseName)
+    case None => throw new RuntimeException("LDBC_TEST_ROOT is not set!")
+  }
+  private val archivePath: Path = resourcesDir.resolve(s"${caseName}.tar.zst")
+  //  Unreachable vertices should be given the value infinity (represented as 9223372036854775807).
+  val UNREACHABLE_ID_VALUE = 9223372036854775807L
+
+  private def checkZSTD(): Unit = {
+    try {
+      s"zstd --version".!
+    } catch {
+      case e: Exception =>
+        throw new RuntimeException(
+          "zstd is not available or not found. Please install zstd and try again.",
+          e)
+    }
+  }
+  def downloadLDBCIfNotExists(): Unit = {
+    if (Files.notExists(resourcesDir) || (Files.list(resourcesDir).count() == 0L)) {
+      println("LDBC data not found. Downloading...")
+      checkZSTD()
+      if (Files.notExists(resourcesDir)) {
+        Files.createDirectory(resourcesDir)
+      }
+      val connection = LDBCUrl.openConnection()
```

Comment:
We’re doing this manually to avoid a dependency, I take it?

Reply:
@rjurney For me, the perfect solution would be to grate the repository `graphframes/graphframes-testing`, place all the LDBC data here, add downloading scripts and use it as a git submodule in main graphframes. What do you think about it?